dotCMS/src/main/java/com/dotmarketing/portlets/cmsmaintenance/action/ViewCMSMaintenanceAction.java

```diff
@@ -479,7 +547,339 @@ private void moveAssetsToBackupDir() throws FileNotFoundException, IOException{
+				/* we will only export 10,000,000 items of any given type */
+				for(i=0;i < 10000000;i=i+step){
+
+					_dh = new HibernateUtil(clazz);
+					_dh.setFirstResult(i);
+					_dh.setMaxResults(step);
+
+					//This line was previously like;
+					//_dh.setQuery("from " + clazz.getName() + " order by 1,2");
+					//This caused a problem when the database is Oracle because Oracle causes problems when the results are ordered
+					//by an NCLOB field. In the case of dot_containers table, the second field, CODE, is an NCLOB field. Because of this,
+					//ordering is done only on the first field for the tables, which is INODE
+					if(com.dotmarketing.beans.Tree.class.equals(clazz)){
+						_dh.setQuery("from " + clazz.getName() + " order by parent, child, relation_type");
+					}
+					else if(MultiTree.class.equals(clazz)){
+						_dh.setQuery("from " + clazz.getName() + " order by parent1, parent2, child, relation_type");
+					}
+					else if(TagInode.class.equals(clazz)){
+						_dh.setQuery("from " + clazz.getName() + " order by inode, tag_id");
+					}
+					else if(Tag.class.equals(clazz)){
+						_dh.setQuery("from " + clazz.getName() + " order by tag_id, tagname");
+					}
+					else if(CalendarReminder.class.equals(clazz)){
+						_dh.setQuery("from " + clazz.getName() + " order by user_id, event_id, send_date");
+					}
+					else if(Identifier.class.equals(clazz)){
+						dc = new DotConnect();
+						dc.setSQL("select * from identifier order by parent_path, id")
+								.setStartRow(i).setMaxRows(step);
+					} else {
+						_dh.setQuery("from " + clazz.getName() + " order by 1");
+					}
+
+					if(Identifier.class.equals(clazz)){
+						_list = TransformerLocator
+								.createIdentifierTransformer(dc.loadObjectResults()).asList();
+					}else{
+						_list = _dh.list();
+					}
+
+					if(_list.size() ==0){
+						try {
+							_bout.close();
+						}
+						catch( java.lang.NullPointerException npe){}
+						_bout = null;
+
+						break;
+					}
+
+					if(_list != null && _list.size() > 0 && _list.get(0) instanceof Comparable){
+						java.util.Collections.sort(_list);
+					}
+
+					_writing = new File(backupTempFilePath + "/" + clazz.getName() + "_" + formatter.format(i) + ".xml");
+					_bout = new BufferedOutputStream(Files.newOutputStream(_writing.toPath()));
+
+					total = total + _list.size();
+
+					try {
+						Thread.sleep(10);
+					} catch (InterruptedException e) {
+						Logger.warn(this, "An error ocurred trying to create XML files");
+						Logger.error(this,e.getMessage(),e);
+					}
+
+					try {
+						_xstream.toXML(_list, _bout);
+					} finally {
+						_bout.close();
+					}
+
+					_bout = null;
+
+				}
+				Logger.info(this, "writing : " + total + " records for " + clazz.getName());
+			}
+
+
+			/* Run Liferay's Tables */
+			/* Companies */
+			_list = PublicCompanyFactory.getCompanies();
+			List<Company> companies = new ArrayList<Company>(_list);

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace the type specification in this constructor call with the diamond operator ("<>"). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2293)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unused "companies" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1481)